rmate

```diff
@@ -381,11 +390,32 @@ function handle_connection {
 
 # connect to textmate and send command
 #
-exec 3<> "/dev/tcp/$host/$port"
+if [ -n "$unix" ] ; then
+    if [ ! -S "$unix" ] ; then
+        echo "$unix is not a socket!"
+        exit 1
+    fi
+
+    fifodir=$(mktemp -d) && \
+        mkfifo "$fifodir/in" "$fifodir/out" && \
+        ( (nc -U "$unix" < "$fifodir/in" > "$fifodir/out" &) &) && \
+        exec 4> "$fifodir/in" && \
+        exec 3< "$fifodir/out"
 
-if [ $? -gt 0 ]; then
-    echo "Unable to connect to TextMate on $host:$port"
-    exit 1
+    connect_status=$?
+
+    [ -d "$fifodir" ] && rm -rf "$fifodir"
+
+    if [ $connect_status -gt 0 ] ; then
+        echo "Unable to connect to TextMate on $unix"
+        exit 1
+    fi
+else
+    exec 3<> "/dev/tcp/$host/$port" && exec 4>&3
```

Comment:
Unfortunately I wasn't able to figure out how to get read and write to the socket happening on the same file descriptor, so I had to split them, and that meant splitting them even in the TCP case :(